apps/src/lab2/views/components/Instructions/NavigationArea.tsx

```diff
@@ -116,56 +122,70 @@ const NavigationArea: React.FC<NavigationAreaProps> = ({
     }
   }, [validationMessage, isRunning]);
 
-  const continueButtonIsEnabled = useMemo(() => {
-    if (isPredictLevel) {
-      return predictResponseSubmitted;
-    } else if (hasValidationConditions) {
-      return validationSatisfied;
-    } else if (requireRun) {
-      return hasRun;
-    } else if (submittable && hasSubmitted) {
-      return true;
-    } else {
-      return true;
+  // Determines if the submit button should be enabled and what tooltip to show if any.
+  const [submitEnabled, submitTooltip] = useMemo(() => {
+    if (!submittable) {
+      return [false, undefined] as const;
+    }
+    if (hasSubmitted) {
+      return [true, undefined] as const;
     }
+    if (hasValidationConditions && !validationSatisfied) {
+      return [false, lab2I18n.toSubmitValidate()] as const;
+    }
+    if (!disableEditRunForSubmission && (!hasRun || !hasEdited)) {
+      return [false, lab2I18n.toSubmitEditRun()] as const;
```

Comment:
I think we're missing the case when we're in a lab like `weblab2` and the user needs to edit code but not run in order to submit for the most accurate tooltip msg.

Reply:
Oh good call - will add!